src/language-server/project/base.ts

```diff
@@ -131,6 +133,14 @@ export abstract class GraphQLProject {
   abstract onCodeLens?: ConnectionHandler["onCodeLens"];
   abstract onCodeAction?: ConnectionHandler["onCodeAction"];
 
+  abstract onUnhandledRequest?: StarRequestHandler;
+  abstract onUnhandledNotification?: (
+    connection: Connection,
```

Comment:
Deviating from the `StarNotificationHandler` signature and prepending `connection`, so the handler could actually do something in response to the notification if necessary.
(Right now I'm just using these handlers for logging though, this is more of a future-proofing)